apps/web/src/components/activity/ExecutionDetailsModal.tsx

```diff
@@ -1,10 +1,77 @@
-import { Group, Modal, useMantineTheme } from '@mantine/core';
-import { colors, shadows, Title, Text } from '../../design-system';
+import { Grid, LoadingOverlay, Modal, useMantineTheme } from '@mantine/core';
+import { useQuery } from 'react-query';
+import { Link } from 'react-router-dom';
+import styled from 'styled-components';
+
+import { ExecutionDetailsSteps } from './ExecutionDetailsSteps';
 
+import { getActivityForNotification } from '../../api/activity';
+import { colors, shadows, Title, Text } from '../../design-system';
+import { ArrowLeft } from '../../design-system/icons';
 import { GotAQuestionButton } from '../utils/GotAQuestionButton';
 
-export function ExecutionDetailsModal({ modalVisibility, onClose }: { modalVisibility: boolean; onClose: () => void }) {
+const LinkText = styled(Text)`
+  color: ${colors.B60};
+  font-size: 14px;
+  line-height: 17px;
+  padding-left: 5px;
+  padding-top: 3px;
+`;
+
+const LinkWrapper = styled.div`
+  display: flex;
+  justify-content: start;
+  padding-top: 35px;
+`;
+
+const ActionsWrapper = styled(LinkWrapper)`
+  margin: 0;
+  padding: 0;
+`;
+
+const ExecutionDetailsFooter = ({ onClose, origin }) => {
+  // TODO: Might be a good idea pass the name of the origin rather than the location path
+  const linkText = `Back to ${origin.replace('/', '')}`;
+
+  return (
+    <Grid gutter={10}>
+      <Grid.Col span={3}>
+        <LinkWrapper>
+          <ArrowLeft height={24} width={24} color={colors.B60} />
+          <Link to={origin} onClick={onClose}>
+            <LinkText>{linkText}</LinkText>
+          </Link>
+        </LinkWrapper>
+      </Grid.Col>
+      <Grid.Col span={2} offset={7}>
+        <ActionsWrapper>
+          {/* TODO: Button has a margin top that's not possible to overload */}
+          <GotAQuestionButton mt={30} size="md" />
+        </ActionsWrapper>
+      </Grid.Col>
+    </Grid>
+  );
+};
+
+export function ExecutionDetailsModal({
+  notificationId,
+  modalVisibility,
+  origin,
+  onClose,
+}: {
+  notificationId: string;
+  modalVisibility: boolean;
+  onClose: () => void;
+  origin: string;
+}) {
   const theme = useMantineTheme();
+  const { data, isLoading, isFetching } = useQuery(['activity', 'notificationId'], function () {
```

Comment:
```suggestion
  const { data, isLoading, isFetching } = useQuery(['activity', notificationId], function () {
```

Reply:
It should be like this and then you should not get cached results when you are viewing different activities 😄